gluten-core/src/main/scala/io/glutenproject/expression/ArrayExpressionTransformer.scala

```diff
@@ -86,3 +91,101 @@ class GetArrayItemTransformer(substraitExprName: String, left: ExpressionTransfo
     ExpressionBuilder.makeScalarFunction(functionId, exprNodes, typeNode)
   }
 }
+
+class SequenceTransformer(substraitExprName: String, start: ExpressionTransformer,
+  stop: ExpressionTransformer, stepOpt: Option[ExpressionTransformer], original: Sequence)
+    extends ExpressionTransformer
+    with Logging {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    // In Spark: sequence(start, stop, step)
+    // In CH: if ((end - start) % step = 0, range(start, end + step, step), range(start, end, step))
```

Comment:
@rui-mo please check whether there is the same as the velox ?

Reply:
hi @zzcclp, Velox does not have this function yet. But I have a feeling that we'd better generate the Substrait plan with compatible signature as Spark, and it's the duty of each backend to conduct any special handling in the SubstraitToVelox conversion. How do you think?

Reply:
Actually sometimes it is more easy and clear to add special handling during spark -> substrait plan. I don't think it is an issue since we can distinguish different backends in gluten.

Reply:
It's OK to me.